src/aws/osml/model_runner/tile_worker/variable_overlap_tiling_strategy.py

```diff
@@ -0,0 +1,251 @@
+#  Copyright 2024 Amazon.com, Inc. or its affiliates.
+
+import logging
+from math import ceil, floor
+from typing import Dict, List, Tuple
+
+from geojson import Feature
+
+from ..common import ImageDimensions, ImageRegion, get_feature_image_bounds
+from ..inference import FeatureSelector
+from .tiling_strategy import TilingStrategy, ceildiv, generate_crops
+
+logger = logging.getLogger(__name__)
+
+
+class VariableOverlapTilingStrategy(TilingStrategy):
+    def compute_regions(
+        self,
+        processing_bounds: ImageRegion,
+        region_size: ImageDimensions,
+        tile_size: ImageDimensions,
+        overlap: ImageDimensions,
+    ) -> List[ImageRegion]:
+        """
+        Identify the regions that should be created from this image.
+
+        :param processing_bounds: the bounds of the full image or area of interest in pixels ((r, c), (w, h))
+        :param region_size: the size of the regions in pixels (w, h)
+        :param tile_size: the size of the tiles in pixels (w, y)
+        :param overlap: the amount of overlap (w, h)
+
+        :return: a collection of region boundaries
+        """
+        logger.debug(
+            f"ENTER FixedTileTilingStrategy.compute_regions: {processing_bounds}, {region_size}, {tile_size}, {overlap}"
+        )
+
+        adjusted_overlap = self._calculate_overlap_for_full_tiles(processing_bounds[1], tile_size, overlap)
+        logger.debug(f"FixedTileTilingStrategy.compute_regions: adjusted_overlap = {adjusted_overlap}")
```

Comment:
VariableOverlapTilingStrategy

Reply:
also in other logging statements but won't mark each one individually.

Reply:
fixed